### PR TITLE
Improve documentation for github actions

### PR DESCRIPTION
**Pull request by another-rex**
Attempt at the 3 remaining items mentioned in https://github.com/google/osv-scanner/issues/516. 

Feel free to do any rewording or rearranging @hayleycd to make things clearer!

Also fix the SARIF test since it was still calling `createTextFixture` instead of `assertTextFixture`

### CODE REVIEW

**Comment by hayleycd** [comment]
Can you build the docs from this branch so we can preview them? @another-rex

File: docs/github-action.md
Comment (hayleycd) [inline docs/github-action.md:17]: Maybe we want to rename this section, as we now run scans on either releases or on a schedule.

File: docs/github-action.md
Comment (hayleycd) [inline docs/github-action.md:14]: Hey @another-rex . 

I tried to add commits to the PR but I don't think I have the right permissions, so I am going to comment instead. 

Please either:
1. Make these changes before merging
2. Update permissions so I can add commits. I tried to follow [these instructions] (https://tighten.com/insights/adding-commits-to-a-pull-request/) and got an error message that said "fatal: Could not read from remote repository.Please make sure you have the correct access rights and the repository exists." 
3. Merge it anyway and I can do another PR with my updates. Nothing is wrong, just not optimal so it would be ok to merge.

Comment (hayleycd) [inline docs/github-action.md:14]: It isn't good practice that we have the scheduled scan and the release scan grouped together in this first item in the list and then they aren't grouped that way in the instructions below which are in this order:
1. Scheduled scan
2. PR scan
3. Release scan

Would you prefer to update the order of the instructions, or how they are grouped in lines 14 & 15?

File: docs/github-action.md
Comment (hayleycd) [inline docs/github-action.md:52]: I think we may want to be a little more direct about what `osv-scanner-reuseable.yml` is. If I was newer to GitHub Actions I might be confused because I don't need this file in my project.

File: docs/github-action.md
Comment (hayleycd) [inline docs/github-action.md:54]: I think it would also be good practice to explain the defaults.

**Comment by hayleycd** [comment]
The GitHub Action page is getting a little long, I would include a table of contents like some of the other pages have. I can put this in if you would like me to make my changes in a separate PR.

**Review by hayleycd — APPROVED** [review]
I am approving these.

I have commented some changes I would like to see, but they don't necessarily need to go in this PR. Please see the options in [this comment](https://github.com/google/osv-scanner/pull/575/files/c75ac35506e6dbd8b0002b502a89f86a3bc0cae5#r1349755229) and let me know what you prefer.

**Comment by another-rex** [comment]
Should now be previewing this branch here: https://another-rex.github.io/osv-scanner/

**Comment by another-rex** [comment]
@hayleycd Updated your permissions to maintainer, you should be able to push to PR branches now.

File: docs/github-action.md
Comment (another-rex) [reply docs/github-action.md:14]: That makes sense, let's change it to this order so that scheduled and release is together:

1. PR scan
2. Scheduled scan
3. Release scan

Should we still separate out the two entries on line 14&15? We'll probably have to reword it to something like 
> We currently support 3 different ways to use Github Actions:
>
> 1. PR Scan ...
> 2. Scheduled Scan ...
> 3. Release Scan ...
>

Comment (another-rex) [reply docs/github-action.md:14]: Thanks for the comments! Added permissions to allow you to push, do you want me to make the changes you proposed, or would you like to push the changes on your side, and I'll just review them?

File: docs/github-action.md
Comment (another-rex) [reply docs/github-action.md:52]: Good point, maybe adding something like

> In your github action file (`.github/workflows/osv-scanner-scheduled.yml`)...

could point the user at the file they are meant to edit.

File: docs/github-action.md
Comment (another-rex) [reply docs/github-action.md:54]:
```suggestion
Importantly `--format` and `--output` flags are already set by the reusable workflow and should not be overridden here.

The default behavior will recursively scan your repository from the root, but skip commit scanning to focus on dependencies.
```

File: docs/github-action.md
Comment (another-rex) [reply docs/github-action.md:14]: Updated the ordering in f62f6c3

File: docs/github-action.md
Comment (another-rex) [reply docs/github-action.md:54]: Added defaults value as a code block, and added inline comments describing what each line does

**Comment by hayleycd** [comment]
@another-rex , I tried to add commits again with no luck. I think it would be best to merge this, and I will follow up with a few small changes in a PR tomorrow.

**Pull request merged**
